pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -117,14 +114,11 @@
 , WebKit
 }:
 
-assert (withGTK2 && !withNS && variant != "macport") -> withX;
 assert (withGTK3 && !withNS && variant != "macport") -> withX || withPgtk;
 
-assert noGui -> !(withX || withGTK2 || withGTK3 || withNS || variant == "macport");
+assert noGui -> !(withX || withGTK3 || withNS || variant == "macport");
 assert withAcl -> stdenv.isLinux;
 assert withAlsaLib -> stdenv.isLinux;
-assert withGTK2 -> !(withGTK3 || withPgtk);
-assert withGTK3 -> !withGTK2 || withPgtk;
```

Comment:
Shouldn't it be `assert withGTK3 -> withPgtk;`?

Reply:
I don't think so.

For all the effects and purposes, removing GTK2 support renders `withGtk2` as `false`.

Therefore,

```nix
assert withGTK2 -> !(withGTK3 || withPgtk); 
assert withGTK3 -> !withGTK2 || withPgtk;
```

renders as

```nix
assert false -> !(withGTK3 || withPgtk); 
assert withGTK3 -> true || withPgtk;
```
effectively no-op.

Further, `assert withGTK3 -> withPgtk;` would be problematic: GTK3 libraries are used in both GTK3-X11 and PGTK (the Wayland-friendly) frontends.
But specific libs and flags for PGTK are not compatible with GTK3-X11. (Indeed, using emacs-pgtk in a X11 wm emits a warning.)

On the other hand, the reversed `assert withPgtk -> withGTK3;` is already covered by the line 124, ` assert withPgtk -> withGTK3 && !withX;`.